alpha/declcfg/declcfg_to_model.go

```diff
@@ -168,6 +168,70 @@ func ConvertToModel(cfg DeclarativeConfig) (model.Model, error) {
 		}
 	}
 
+	// deprecationsByPackage tracks the set of package names
+	// and is used to detect duplicate packages.
+	deprecationsByPackage := sets.New[string]()
+
+	for i, deprecation := range cfg.Deprecations {
+
+		// no need to validate schema, since it could not be unmarshaled if missing/invalid
+
+		if deprecation.Package == "" {
+			return nil, fmt.Errorf("package name must be set for deprecation item %v", i)
+		}
+
+		// must refer to package in this catalog
+		mpkg, ok := mpkgs[deprecation.Package]
+		if !ok {
+			return nil, fmt.Errorf("cannot apply deprecations to an unknown package %q", deprecation.Package)
+		}
+
+		// must be unique per package
+		if deprecationsByPackage.Has(deprecation.Package) {
+			return nil, fmt.Errorf("expected a maximum of one deprecation per package: %q", deprecation.Package)
+		}
+		deprecationsByPackage.Insert(deprecation.Package)
+
+		references := sets.New[PackageScopedReference]()
```

Comment:
I see now. Is using the `PackageScopedReference` type required here? Could we have the same functionality using the flat `DeprecationEntry` type by using `sets.New[string]()` and doing some string concatenation of the `DeprecationEntry.Schema` and `DeprecationEntry.Name` fields?

Reply:
We totally could, but it's cleaner if we can acknowledge the composite type instead of doing some behind-the-scenes string concatenation.

Reply:
Also string concatenation is a place for bugs to creep in if you aren't extremely careful and have some validation to make sure that you don't end up with apparent, but not actual duplicates:
- `foo`, `bar` => `foobar`
- `foob`, `ar` => `foobar`

Reply:
Fair enough.